ui/tickets_page.go

```diff
@@ -0,0 +1,477 @@
+	txt := c.theme.Label(values.MarginPadding15, "")
+	txt.MaxLines = 1
+
+	var dateTime, status, hash, amount string
+	var blockHeight int32
+	switch t := ticket.(type) {
+	case wallet.Ticket:
+		info := t.Info
+		blockHeight = info.BlockHeight
+		dateTime, status, hash, amount = t.DateTime, info.Status, info.Ticket.Hash.String(), t.Amount
+	case wallet.UnconfirmedPurchase:
+		blockHeight, dateTime, status, hash, amount = t.BlockHeight, t.DateTime, t.Status, t.Hash, t.Amount
+	}
+
+	return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding60)
+			return c.theme.Label(values.MarginPadding15, fmt.Sprintf("%d", blockHeight)).Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding120)
+			txt.Text = dateTime
+			return txt.Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding120)
+			txt.Text = status
+			return txt.Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding150)
+			txt.Text = fmt.Sprintf("%s...%s", hash[:8], hash[56:])
+			return txt.Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding150)
+			txt.Alignment = text.End
+			txt.Text = amount
+			return txt.Layout(gtx)
+		}),
+	)
+}
+
+func (pg *ticketPage) purchaseTicket(c pageCommon, password []byte) {
+	selectedWallet := c.info.Wallets[pg.walletsDropdown.SelectedIndex()]
+	selectedAccount := selectedWallet.Accounts[pg.accountsDropdown.SelectedIndex()]
+
+	pg.vspd = c.wallet.NewVSPD(selectedWallet.ID, selectedAccount.Number)
+	_, err := pg.vspd.GetInfo()
+	if err != nil {
+		log.Error("[GetInfo] err:", err)
+		return
+	}
+
+	numbTicketsStr := pg.inputNumberTickets.Editor.Text()
+	numbTickets, err := strconv.Atoi(numbTicketsStr)
+	if err != nil {
+		c.Notify(err.Error(), false)
+		return
+	}
+
+	expiryBlocksStr := pg.inputExpiryBlocks.Editor.Text()
+	expiryBlocks, err := strconv.Atoi(expiryBlocksStr)
+	if err != nil {
+		c.Notify(err.Error(), false)
+		return
+	}
+
+	hash, err := c.wallet.PurchaseTicket(selectedWallet.ID, selectedAccount.Number, uint32(numbTickets), password, uint32(expiryBlocks))
+	if err != nil {
+		log.Error("[PurchaseTicket] err:", err)
+		c.Notify(err.Error(), false)
+		return
+	}
+
+	transactionResponse, err := pg.getTicketFee(hash, password)
+	if err != nil {
+		log.Error("[GetTicketFee] err:", err)
+		c.Notify(err.Error(), false)
+		return
+	}
+
+	_, err = pg.vspd.PayVSPFee(transactionResponse, hash, "", password)
+	if err != nil {
+		log.Error("[PayVSPFee] err:", err)
+		c.Notify(err.Error(), false)
+		return
+	}
+
+	c.Notify("success", true)
+	pg.inputExpiryBlocks.Editor.SetText("")
+	pg.inputNumberTickets.Editor.SetText("")
+	c.closeModal()
+}
+
+func (pg *ticketPage) getTicketFee(ticketHash string, password []byte) (feeTransaction string, err error) {
```

Comment:
The method can be renamed to "createTicketFeeTransaction" since what it gets the fee, creates a transaction and returns it. However, methods or functions aren't just for reuse. They're also pretty useful for keeping code clean, easy to read and organized. As is evident in the layout code when we define sections as methods even though they might not be reused. That was the goal for putting ticket transaction calls in a separate method.

Reply:
Renaming should work.